pkg/controller/workload/workload_hash.go

```diff
@@ -25,25 +25,38 @@ var (
 	hash = fnv.New32a()
 )
 
+const tombstoneMarker = "\x00"
+
 // HashName converts a string to a uint32 integer as the key of bpf map
 type HashName struct {
 	numToStr map[uint32]string
+	// records its tombstone number given a string
+	tombstones map[string]uint32
```

Comment:
This is kind of mem leak. Our requirement is:

1. a str should be allocated with a consistent value
2. after it deleted, and then at some time recreated, the value can be different.
3. It's better to be consistent for the case that kmesh daemon restart, which seems difficult with this implement. Maybe we can persist the maps somewhere.

Reply:
> 2\. after it deleted, and then at some time recreated, the value can be different.

Requirement 1 means for a str like "foo", if we call StrToNum we will get number 123. Then at anytime we StrToNum("foo") should always get 123? Even if collision happened?
Can you give an example about requirement 2?

Reply:
In the original implemention, there exists another problem.
If A collides with B.
If StrToNum(A) = 123, then StrToNum(B) = 124
Then we delete A, we will delete numToStr[123].
Then we delete B, we will also delete numToStr[123], cause Delete will call StrToNum, where StrToNum(B) will be 123, if A is deleted(since there is no collision any more). In this case, we will leave numToStr[124] = B, which is also a kind of mem leak.

Reply:
>after it deleted, and then at some time recreated, the value can be different.


say `foo` at t0 is allcated `1234`, then at t1, it is deleted. In our case, all the info is deleted from bpf map.

Maybe t2, `foo` is recreated, then we can allocate it to `2345`.  It need to be consistent during its lifecycle.

Reply:
I got it. We need to persist the numtostr map in case kmesh restarts.

Reply:
> In the original implemention, there exists another problem. If A collides with B. If StrToNum(A) = 123, then StrToNum(B) = 124 Then we delete A, we will delete numToStr[123]. Then we delete B, we will also delete numToStr[123], cause Delete will call StrToNum, where StrToNum(B) will be 123, if A is deleted(since there is no collision any more). In this case, we will leave numToStr[124] = B, which is also a kind of mem leak.

to solve this problem, we will still need the  tombstone marker

Reply:
We only need to store string that cause collision.
For example, if A collides with B.
We call StrToNum(A), then StrToNum(B).
A got "123", B got "124"
We only need to persist (B, 124). Since after restart, we call StrToNum(B) we can get StrToNum(B) = 124 from persisted map. And StrToNum(A) = 123 according to hash value.